src/core/nodeOps.ts

```diff
@@ -23,6 +23,17 @@ const supportedPointerEvents = [
   'onPointerLeave',
 ]
 
+export function invalidateInstance(instance: TresObject) {
+  const ctx = instance.userData.tres__root?.userData?.tres__context
+  
+  if (!ctx) return
+  
+  if (ctx.renderMode.value === 'on-demand' && ctx.internal.frames.value === 0) {
```

Comment:
The conditional would probably be better handled privately within `ctx`, as in the ["Tell, don't ask" principle](https://martinfowler.com/bliki/TellDontAsk.html).

As it stands, changes to `ctx`'s implementation, or the renderMode's frames logic might require that this be updated as well.